### PR TITLE
🐛 Issue 698: Search Bug in the Emojipedia Play 

### LINKED ISSUE #698: 🐛 [Bug report]: Search Bug in the Emojipedia Play
### Descrcibe the bug?

in the emojipedia search
- if I write a word that results in no emoji found even if I backspace and reach to a query word where i should be shown some emojis but it doesn't until I completely clear the search

### Steps to reproduce the bug?

1. Go to Emojipedia play
2. type some random text in saerch
3. then Backspace 
4. You won't see the emojis until you completely clear the search

### Expected behavior

update on backspace as well.

### Desktop (Please provide your system information)

Browser: Chrome

### Mobile (Please provide your device information)

_No response_

### Screenshot / Screenshare

image
image

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by nishasen**
> First thing, PLEASE READ THIS: [ReactPlay Code Review Checklist](https://github.com/reactplay/react-play/wiki/ReactPlay-Code-Review-Checklist)

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #698 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Issue : According to the bug, the search was happening only after removing the whole word once searched.

Reason : This was happening because the search was being made from the previous search result.

Solution : I have changed the array from which the search was happening, that is, from the whole data every time, instead of searching from the previous data.

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings

### CODE REVIEW

**Comment by atapas** [comment]
@Angryman18 please review this one.

**Comment by Angryman18** [comment]
i see the api is failing to give data back. so i think some error handling is needed. coz even if api fails it shouldn't end up into error boundary rather we might want to put a proper error message?
Screenshot from 2022-10-15 22-57-15

File: src/plays/emojipedia/Emojipedia.js [hunk 1]
```diff
@@ -32,8 +32,7 @@ function Emojipedia(props) {
   useEffect(() => {
     const delayFn = setTimeout(() => {
       if (query) {
-        setEmojisList((prevEmojiList) =>
-          prevEmojiList?.filter((emoji) =>
+        setEmojisList(data.filter((emoji) =>
```
Comment (Angryman18) [inline src/plays/emojipedia/Emojipedia.js:35]: i see more than 2.5k emojis are coming from the api. please use `useTransition` hook here to make the filter.

**Review by Angryman18 — CHANGES_REQUESTED** [review]

File: src/plays/emojipedia/Emojipedia.js (hunk 1 above)
Comment (nishasen) [reply src/plays/emojipedia/Emojipedia.js:35]: Okay working on that, I have not used useTransition before, but as I read its kind of loading, will make the necessary changes and upload

Comment (nishasen) [reply src/plays/emojipedia/Emojipedia.js]: @Angryman18 can you please review the code again I have made the changes suggested

Comment (Angryman18) [reply src/plays/emojipedia/Emojipedia.js]: on deployed url i still see the page break and api call error. although if the api is not meant to give data on the deployed url then handle the error on api fail
Screenshot from 2022-10-17 12-27-22

**Comment by atapas** [comment]
@Angryman18 This works now on preview... Please finish the review and fixes for merge.

**Review by Angryman18 — APPROVED** [review]
LGTM

**Review by atapas — APPROVED** [review]

**Review by atapas — APPROVED** [review]
Approved

**Comment by atapas** [comment]
@all-contributors please add @nishasen for Code

**Pull request merged**